plugins/experimental/alt_svc/single_service_file_map.cc

```diff
@@ -0,0 +1,118 @@
+/**
+  @file
+  @brief TODO documentation
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include "ip_host_map.h"
+#include "prefix_parser.h"
+#include <fstream>
+#include <algorithm>
+
+using namespace std;
+
+string
+SingleServiceFileMap::findHostForIP(const sockaddr *ip, string hostname) const noexcept
+{
+  return this->findHostForIP(ip);
+}
+
+string
+SingleServiceFileMap::findHostForIP(const sockaddr *ip) const noexcept
+{
+  char *output;
+  // this->print_the_map();
+  return this->host_map.contains(ip, (void **)&output) ? output : "";
+}
+
+// Lifted from ControlMatcher.cc
+void
+SingleServiceFileMap::print_the_map() const noexcept
+{
+  printf("\tIp Matcher with %zu ranges.\n", this->host_map.getCount());
+  for (IpMap::iterator spot(this->host_map.begin()), limit(this->host_map.end()); spot != limit; ++spot) {
+    char b1[INET6_ADDRSTRLEN], b2[INET6_ADDRSTRLEN];
+    printf("\tRange %s - %s ", ats_ip_ntop(spot->min(), b1, sizeof b1), ats_ip_ntop(spot->max(), b2, sizeof b2));
+    printf("Host: %s \n", static_cast<char *>(spot->data()));
+  }
+}
+
+SingleServiceFileMap::SingleServiceFileMap(string filename)
+{
+  // Read file
+  int fail = 0;
+  ifstream config_file{filename};
+  if (config_file.fail()) {
+    // TODO I don't know how to get this to link for tests
+    // TS_DEBUG(PLUGIN_NAME, "Cannot find a config file at: %s", filename.c_str());
```

Comment:
It's defined in ts.h. It works for prefix_parser, but I don't know what I'm doing wrong with including/linking that makes it not work for ip_host_map test